ilc/server/services/CanonicalTagService.js

```diff
@@ -0,0 +1,16 @@
+const IlcIntl = require('ilc-sdk/app').IlcIntl;
+const { context } = require('../context/context');
+const { removeQueryParams } = require('../../common/utils');
+
+class CanonicalTagService {
+    static getCanonicalTagForUrlAsHTML(url, locale, i18nConfig) {
+        const store = context.getStore();
+        const fullUrl = removeQueryParams(`${store.get('protocol')}://${store.get('domain')}${url}`);
```

Comment:
seems that might have XSS here, please double check that it is not possible through crafting malicious URL

Reply:
Interesting question. How I thought about it.
`protocol` is read from request state
`domain` as well
`url` is processed by i18n and passed after preliminary processing. 

Looking on other cases

Reply:
We are using new URL interface under the hood so I expect internal sanitizing is enough to ensure URL is valid. If URL is valid it acts as safe URL.